SelfAndPeerEvaluations/partner_eval_quiz_creator.py

```diff
@@ -0,0 +1,161 @@
+
+
+class PartnerEvalQuizCreator:
+    def __init__(
+        self,
+        course: canvasapi.course.Course,
+        json_questions: dict,
+        assignment_name: str,
+        unlock_date: Optional[datetime.datetime] = None,
+        due_date: Optional[datetime.datetime] = None,
+        lock_date: Optional[datetime.datetime] = None,
+        assignment_group_id: int = 0,
+    ):
+        """
+        public functions
+        ---
+        upload_to_canvas(): uploads a evaluation quiz to canvas
+        """
+        self.course = course
+        self.json_questions = json_questions
+        self.assignment_name = assignment_name
+        self.assignment_group_id = (
+            assignment_group_id
+            if assignment_group_id
+            else self.__find_partner_eval_ag_id(course)
+        )
+        self.unlock_date = unlock_date
+        self.due_date = due_date
+        self.lock_date = lock_date
+
+    def upload_to_canvas(self) -> None:
+        print("Uploading quiz assignment to canvas")
+        quiz_info = self.__create_quiz_info()
+        quiz_questions = self.__create_quiz_questions()
+
+        quiz = self.course.create_quiz(quiz_info)
+        for quiz_question in quiz_questions:
+            quiz.create_question(question=quiz_question)
+
+        canvas_assignment = course.get_assignment(quiz.assignment_id)
+        canvas_assignment.edit(
+            assignment={"omit_from_final_grade": True, "published": True}
+        )
+
+        print("Finished upload!")
+
+    def __create_quiz_info(self) -> dict:
+        return {
+            "title": self.assignment_name,
+            "description": self.json_questions["description"],
+            "quiz_type": self.json_questions["quiz_type"],
+            "assignment_group_id": self.assignment_group_id,
+            "scoring_policy": self.json_questions["scoring_policy"],
+            "published": self.json_questions["published"],
+            "show_correct_answers": self.json_questions["show_correct_answers"],
+            "unlock_at": self.unlock_date,
+            "due_at": self.due_date,
+            "lock_at": self.lock_date,
+            "allowed_attempts": self.json_questions["allowed_attempts"],
+        }
+
+    def __create_quiz_questions(self) -> list[dict]:
+        """
+        Iterates through all the questions in the JSON file's question list
+        Most of the questions in the JSON file are properly formated formated
+        for canvasapi, but some have missing fields (answer section, pts possible)
+        :return: list of questions ready to send to canvasapi's create_question()
+        """
+        quiz_questions = []
+        multiple_choice_answers = dict(
+            enumerate(
+                {"answer_text": choice, "answer_weight": 1}
+                for choice in [
+                    "Strongly disagree",
+                    "Disagree",
+                    "Agree",
+                    "Strongly agree",
+                ]
```

Comment:
I know when we talked live, you were against moving the choices into the questions json file as  it made it too bloated. I still think that these data options should be moved out of the code as I think editing them in one central area will be easier instead of multiple places across the program. I was thinking that maybe we might a defaults_value.json that controls what the default values are. If that file gets too big we could split it into a defaults directory that has multiple jsons in it, one for each type of thing that needs defaults.